SerialPrograms/Source/Integrations/DiscordSocial/DiscordSocial.cpp

```diff
@@ -0,0 +1,171 @@
+#ifdef PA_SOCIAL_SDK
+#define DISCORDPP_IMPLEMENTATION
+
+#include <thread>
+#include "DiscordSocial.h"
+#include "CommonFramework/Logging/Logger.h"
+#include "Integrations/ProgramTracker.h"
+#include "../../CommonFramework/GlobalSettingsPanel.h"
+#include "../../CommonFramework/Globals.h"
+
+using namespace discordpp;
+namespace PokemonAutomation{
+namespace Integration{
+namespace DiscordSocialSDK{
+
+DiscordSocial& DiscordSocial::instance(){
+    static DiscordSocial instance;
+    return instance;
+}
+
+void DiscordSocial::run(){
+    m_client = std::make_shared<Client>();
+    if (!m_client){
+        log("Failed to initialize DiscordSocialSDK.", "run()", LoggingSeverity::Error);
+        m_client.reset();

```

Comment:
This isn't exception safe. If an exception is thrown further down, `m_client` remains initialized which is probably not what you intended.

Make the client first and store into a local variable:
```
std::shared_ptrClient> client = std::make_shared<Client>();
```
Then at the the end of the function (before you start the thread), move it into `m_client`. 

It's possible for the thread creation itself to throw. So to be truly correct, you'll need something like this:

```
m_client = std::move(client);
try{
    std::thread(&DiscordSocial::thread_loop, this).detach();
}catch (...){
    m_client.reset();
    throw;
}
```
But see my other comment about not using `detach()`.